src/ui/molecules/stepper/selector/stepper.selector.spec.ts

```diff
@@ -0,0 +1,333 @@
+      expectedResult: Step[]
+    }) => {
+      describe('When calling function', () => {
+        const result = getStepsWithUpdatedStatus(steps, state)
+
+        test(`Then, result value is ${expectedResult}`, () => {
+          expect(result).toEqual(expectedResult)
+        })
+      })
+    }
+  )
+})
+
+describe('Considering the getStepIndex function', () => {
+  const steps: Step[] = [
+    {
+      id: 'step1',
+      status: 'completed'
+    },
+    {
+      id: 'step2',
+      status: 'disabled'
+    },
+    {
+      id: 'step3',
+      status: 'invalid'
+    },
+    {
+      id: 'step4'
+    },
+    {
+      id: 'step5',
+      status: 'disabled'
+    },
+    {
+      id: 'step6'
+    }
+  ]
+
+  describe.each`
+    steps    | id         | expectedResult
+    ${steps} | ${'step1'} | ${0}
+    ${steps} | ${'step2'} | ${1}
+    ${steps} | ${'step3'} | ${2}
+    ${steps} | ${'step4'} | ${3}
+    ${steps} | ${'step5'} | ${4}
+    ${steps} | ${'step6'} | ${5}
+  `(
+    'Given a value <$steps> and <$id>',
+    ({ steps, id, expectedResult }: { steps: Step[]; id: StepId; expectedResult: StepIndex }) => {
+      describe('When calling function', () => {
+        const result = getStepIndex(steps, id)
+
+        test(`Then, result value is ${expectedResult}`, () => {
+          expect(result).toEqual(expectedResult)
+        })
+      })
+    }
+  )
+})
+
+describe('Considering the getCurrentStepIndex function', () => {
+  const steps: Step[] = [
+    {
+      id: 'step1',
+      status: 'completed'
+    },
+    {
+      id: 'step2',
+      status: 'disabled'
+    },
+    {
+      id: 'step3',
+      status: 'invalid'
+    },
+    {
+      id: 'step4'
+    },
+    {
+      id: 'step5',
+      status: 'disabled'
+    },
+    {
+      id: 'step6'
+    }
+  ]
+
+  const state1: StepperState = {
+    currentStep: 'step3',
+    stepStatus: OrderedMap<StepId, StepStatus>()
+      .set('step1', 'completed')
+      .set('step2', 'disabled')
+      .set('step3', 'completed')
+      .set('step4', 'uncompleted')
+      .set('step5', 'disabled')
+      .set('step6', 'uncompleted'),
+    enabledSteps: ImmutableList(['step1', 'step3', 'step4', 'step6'])
+  }
+
+  const state2: StepperState = {
+    ...state1,
+    currentStep: 'step4'
+  }
+
+  const state3: StepperState = {
+    ...state1,
+    currentStep: 'step6'
+  }
+
+  describe.each`
+    steps    | state     | expectedResult
+    ${steps} | ${state1} | ${2}
+    ${steps} | ${state2} | ${3}
+    ${steps} | ${state3} | ${5}
+  `(
+    'Given a value <$steps> and <$state>',
+    ({
+      steps,
+      state,
+      expectedResult
+    }: {
+      steps: Step[]
+      state: StepperState
+      expectedResult: StepIndex
+    }) => {
+      describe('When calling function', () => {
+        const result = getCurrentStepIndex(steps, state)
+
+        test(`Then, result value is ${expectedResult}`, () => {
+          expect(result).toEqual(expectedResult)
+        })
+      })
+    }
+  )
+})
```

Comment:
Great, it's a good beginning 😎

Let's work together to make those tests more sexier 😍